mixer/pkg/protobuf/yaml/encoder.go

```diff
@@ -63,69 +63,909 @@ func (e *Encoder) EncodeBytes(data map[interface{}]interface{}, msgName string,
 }
 
 func (e *Encoder) visit(name string, data interface{}, field *descriptor.FieldDescriptorProto, skipUnknown bool, buffer *proto.Buffer) error {
+	if data == nil {
+		return nil
+	}
+	repeated := field.IsRepeated()
+	packed := field.IsPacked() || field.IsPacked3()
+	wireType := uint64(field.WireType())
+	fieldNumber := int(field.GetNumber())
+	if packed {
+		wireType = uint64(proto.WireBytes)
+	}
 	switch field.GetType() {
```

Comment:
Nit: You can break this method up for readability. Specifically, each case statement can be a function call.